MainRepository/Program.cs

```diff
@@ -1,7 +1,44 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
+﻿// Implementation of Anagram
+class Anagram {
+
+   public static void Main (string[] args) {
+      Anagram a = new Anagram ();
+      a.test ();
+   }
+
+   // Function to find given two strings are anagram or not
+   bool IsAnagram (string str1, string str2) {
+      if (str1.Length != str2.Length) return false;
+      if (str1 == str2) return true;
+      Dictionary<char, int> word = new Dictionary<char, int> ();
+      foreach (char element in str1.ToCharArray ()) {
+         if (word.ContainsKey (element)) word[element]++;
+         else word.Add (element, 1);
+      }
+      foreach (char element in str2.ToCharArray ()) {
+         if (!word.ContainsKey (element)) return false;
+         if (--word[element] == 0) word.Remove (element);
+      }
+      return word.Count == 0;
+   }
+
+   public void test () {
+      for (int i = 0; i < words.Length; i++) {
+         if (words[i].StartsWith (";")) continue;
+         temp.Add (words[i], new () { words[i] });
+         for (int j = i + 1; j < words.Length; j++) {

```

Comment:
This is cumbersome. For each word you are reading all the words below it.
Instead could have just had the key as the string which is the word with characters sorted in ascending order and in one loop itself you could have got the anagrams.